### ISSUE TITLE
  Testcases failure for PhantomJS2.0

### ISSUE THREAD

**Issue by salamani** [strongloop/loopback#1898]
Hi,
I am having issue of PhantomJS crashing while running the testcases.
The  PHANTOMJS_BIN set to /bin/phantomjs where I do have a  PhantomJS2.0 which has been ported for ppc64le machine and working correctly.
Strongloop should migrate to 2.0.0 because the breakpad code is removed from 2.0.0. Since this is a very complex code to port, 1.9.8 can not be ported to ppc64le easily.

The log while running testcase for karma:unit-once is as follow:

Running "karma:unit-once" (karma) task
[D] Task source: /root/ravi/loopback/node_modules/grunt-karma/tasks/grunt-karma.js
DEBUG [config]: autoWatch set to false, because of singleRun
DEBUG [plugin]: Loading plugin karma-browserify.
DEBUG [plugin]: Loading plugin karma-mocha.
DEBUG [plugin]: Loading plugin karma-phantomjs-launcher.
DEBUG [plugin]: Loading plugin karma-chrome-launcher.
DEBUG [plugin]: Loading plugin karma-junit-reporter.
DEBUG [framework.browserify]: created browserify bundle: /root/ravi/loopback/90275854a59f43cdfd49f868813a09b9.browserify
DEBUG [framework.browserify]: add bundle to config.files at position 3
DEBUG [framework.browserify]: building bundle
DEBUG [framework.browserify]: updating test/support.js in bundle
DEBUG [framework.browserify]: updating test/loopback.test.js in bundle
DEBUG [framework.browserify]: updating test/model.test.js in bundle
DEBUG [framework.browserify]: updating test/geo-point.test.js in bundle
DEBUG [framework.browserify]: updating test/replication.test.js in bundle
DEBUG [framework.browserify]: updating test/change.test.js in bundle
DEBUG [framework.browserify]: updating test/checkpoint.test.js in bundle
DEBUG [framework.browserify]: updating test/app.test.js in bundle
DEBUG [framework.browserify]: bundling
INFO [framework.browserify]: bundle built
DEBUG [watcher]: Resolved files:
        /root/ravi/loopback/node_modules/mocha/mocha.js
        /root/ravi/loopback/node_modules/karma-mocha/lib/adapter.js
        /root/ravi/loopback/node_modules/es5-shim/es5-shim.js
        /root/ravi/loopback/90275854a59f43cdfd49f868813a09b9.browserify
        /root/ravi/loopback/test/support.js
        /root/ravi/loopback/test/loopback.test.js
        /root/ravi/loopback/test/model.test.js
        /root/ravi/loopback/test/geo-point.test.js
        /root/ravi/loopback/test/replication.test.js
        /root/ravi/loopback/test/change.test.js
        /root/ravi/loopback/test/checkpoint.test.js
        /root/ravi/loopback/test/app.test.js
INFO [karma]: Karma v0.12.37 server started at http://localhost:9876/
INFO [launcher]: Starting browser PhantomJS
DEBUG [temp-dir]: Creating temp dir at /root/ravi/loopback/karma-89306115
DEBUG [launcher]: /bin/phantomjs /root/ravi/loopback/karma-89306115/capture.js
DEBUG [web-server]: serving: /root/ravi/loopback/node_modules/karma/static/client.html
DEBUG [web-server]: serving: /root/ravi/loopback/node_modules/karma/static/karma.js
DEBUG [web-server]: upgrade /socket.io/1/websocket/Kp0iF6V1ftr4l4hqXepN
DEBUG [karma]: A browser has connected on socket Kp0iF6V1ftr4l4hqXepN
INFO [PhantomJS 2.0.0 (Linux 0.0.0)]: Connected on socket Kp0iF6V1ftr4l4hqXepN with id 89306115
DEBUG [launcher]: PhantomJS (id 89306115) captured in 0.631 secs
DEBUG [web-server]: serving: /root/ravi/loopback/node_modules/karma/static/context.html
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/mocha/mocha.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/model.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma-mocha/lib/adapter.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/geo-point.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/replication.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/es5-shim/es5-shim.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/90275854a59f43cdfd49f868813a09b9.browserify
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/support.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/loopback.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/change.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/checkpoint.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/app.test.js
DEBUG [PhantomJS 2.0.0 (Linux 0.0.0)]: Disconnected during run, waiting 10000ms for reconnecting.
DEBUG [launcher]: Process PhantomJS exited with code 0
ERROR [launcher]: PhantomJS crashed.

DEBUG [temp-dir]: Cleaning temp dir /root/ravi/loopback/karma-89306115
INFO [launcher]: Trying to start PhantomJS again (1/2).
DEBUG [launcher]: Restarting PhantomJS
DEBUG [temp-dir]: Creating temp dir at /root/ravi/loopback/karma-89306115
DEBUG [launcher]: /bin/phantomjs /root/ravi/loopback/karma-89306115/capture.js /root/ravi/loopback/karma-89306115/capture.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma/static/client.html
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma/static/karma.js
DEBUG [web-server]: upgrade /socket.io/1/websocket/xWs4OD14gBdwrNHKXepO
DEBUG [karma]: A browser has connected on socket xWs4OD14gBdwrNHKXepO
DEBUG [PhantomJS 2.0.0 (Linux 0.0.0)]: Reconnected on xWs4OD14gBdwrNHKXepO.
WARN [PhantomJS 2.0.0 (Linux 0.0.0)]: Disconnected (1 times), because no message in 30000 ms.
PhantomJS 2.0.0 (Linux 0.0.0): Executed 0 of 0 DISCONNECTED (30.397 secs / 0 secs)
INFO [karma]: Restarting PhantomJS 2.0.0 (Linux 0.0.0) (1 of 1 attempts)
DEBUG [launcher]: Process PhantomJS exited with code 0
DEBUG [temp-dir]: Cleaning temp dir /root/ravi/loopback/karma-89306115
DEBUG [launcher]: Restarting PhantomJS
DEBUG [temp-dir]: Creating temp dir at /root/ravi/loopback/karma-89306115
DEBUG [launcher]: /bin/phantomjs /root/ravi/loopback/karma-89306115/capture.js /root/ravi/loopback/karma-89306115/capture.js /root/ravi/loopback/karma-89306115/capture.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma/static/client.html
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma/static/karma.js
DEBUG [web-server]: upgrade /socket.io/1/websocket/_aNT-FDPsHQyJ-zyXepP
DEBUG [karma]: A browser has connected on socket _aNT-FDPsHQyJ-zyXepP
INFO [PhantomJS 2.0.0 (Linux 0.0.0)]: Connected on socket _aNT-FDPsHQyJ-zyXepP with id 89306115
DEBUG [launcher]: PhantomJS (id 89306115) captured in 31.193 secs
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma/static/context.html
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/mocha/mocha.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/model.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/karma-mocha/lib/adapter.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/geo-point.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/replication.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/node_modules/es5-shim/es5-shim.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/90275854a59f43cdfd49f868813a09b9.browserify
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/support.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/loopback.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/change.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/checkpoint.test.js
DEBUG [web-server]: serving (cached): /root/ravi/loopback/test/app.test.js
DEBUG [PhantomJS 2.0.0 (Linux 0.0.0)]: Disconnected during run, waiting 10000ms for reconnecting.
DEBUG [launcher]: Process PhantomJS exited with code 0
ERROR [launcher]: PhantomJS crashed.
...

Can any please help me in contributing to this issue

Regards,
Salamani

**Comment by superkhau** [comment]
Are you still running into issues? Did you try running the tests with the latest LoopBack?

**Comment by ghatwala** [comment]
@superkhau  ,  i did try with the latest top of the tree for loopback . It still fails with above "PhantmoJS crashed " error.

**Comment by ghatwala** [comment]
@superkhau 
Here is the debug log with latest LoopBack -- >

/**********************************/

Running "karma:unit-once" (karma) task
DEBUG [config]: autoWatch set to false, because of singleRun
DEBUG [plugin]: Loading plugin karma-browserify.
DEBUG [plugin]: Loading plugin karma-mocha.
DEBUG [plugin]: Loading plugin karma-phantomjs-launcher.
DEBUG [plugin]: Loading plugin karma-chrome-launcher.
DEBUG [plugin]: Loading plugin karma-junit-reporter.
DEBUG [framework.browserify]: created browserify bundle: /root/strongloop_pckgs/lp_original/2ae2ca186b3db6f41e52b900ce35dff2.browserify
DEBUG [framework.browserify]: add bundle to config.files at position 3
DEBUG [framework.browserify]: updating test/support.js in bundle
DEBUG [framework.browserify]: building bundle
DEBUG [framework.browserify]: updating test/loopback.test.js in bundle
DEBUG [framework.browserify]: updating test/model.test.js in bundle
DEBUG [framework.browserify]: updating test/geo-point.test.js in bundle
DEBUG [framework.browserify]: updating test/replication.test.js in bundle
DEBUG [framework.browserify]: updating test/change.test.js in bundle
DEBUG [framework.browserify]: updating test/checkpoint.test.js in bundle
DEBUG [framework.browserify]: updating test/app.test.js in bundle
DEBUG [framework.browserify]: bundling
INFO [framework.browserify]: bundle built
DEBUG [watcher]: Resolved files:
        /root/strongloop_pckgs/lp_original/node_modules/mocha/mocha.js
        /root/strongloop_pckgs/lp_original/node_modules/karma-mocha/lib/adapter.js
        /root/strongloop_pckgs/lp_original/node_modules/es5-shim/es5-shim.js
        /root/strongloop_pckgs/lp_original/2ae2ca186b3db6f41e52b900ce35dff2.browserify
        /root/strongloop_pckgs/lp_original/test/support.js
        /root/strongloop_pckgs/lp_original/test/loopback.test.js
        /root/strongloop_pckgs/lp_original/test/model.test.js
        /root/strongloop_pckgs/lp_original/test/geo-point.test.js
        /root/strongloop_pckgs/lp_original/test/replication.test.js
        /root/strongloop_pckgs/lp_original/test/change.test.js
        /root/strongloop_pckgs/lp_original/test/checkpoint.test.js
        /root/strongloop_pckgs/lp_original/test/app.test.js
INFO [karma]: Karma v0.12.37 server started at http://localhost:9876/
INFO [launcher]: Starting browser PhantomJS
DEBUG [temp-dir]: Creating temp dir at /root/strongloop_pckgs/lp_original/karma-47606734
DEBUG [launcher]: /usr/bin/phantomjs /root/strongloop_pckgs/lp_original/karma-47606734/capture.js
â¤â¤â¤â¤â¤DEBUG [web-server]: serving: /root/strongloop_pckgs/lp_original/node_modules/karma/static/client.html
DEBUG [web-server]: serving: /root/strongloop_pckgs/lp_original/node_modules/karma/static/karma.js
DEBUG [web-server]: upgrade /socket.io/1/websocket/Zy6QDy31e3b7HlUXtwuv
DEBUG [karma]: A browser has connected on socket Zy6QDy31e3b7HlUXtwuv
INFO [PhantomJS 2.0.0 (Linux 0.0.0)]: Connected on socket Zy6QDy31e3b7HlUXtwuv with id 47606734
DEBUG [launcher]: PhantomJS (id 47606734) captured in 1.768 secs
DEBUG [web-server]: serving: /root/strongloop_pckgs/lp_original/node_modules/karma/static/context.html
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/node_modules/mocha/mocha.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/node_modules/karma-mocha/lib/adapter.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/geo-point.test.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/model.test.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/2ae2ca186b3db6f41e52b900ce35dff2.browserify
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/loopback.test.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/support.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/node_modules/es5-shim/es5-shim.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/replication.test.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/change.test.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/checkpoint.test.js
DEBUG [web-server]: serving (cached): /root/strongloop_pckgs/lp_original/test/app.test.js
DEBUG [PhantomJS 2.0.0 (Linux 0.0.0)]: Disconnected during run, waiting 10000ms for reconnecting.
DEBUG [launcher]: Process PhantomJS exited with code 0
ERROR [launcher]: PhantomJS crashed.


/*******************************/

My phantomJS version is : 
[root@pts00434-vm7 lp_original]# phantomjs --version
2.0.0
[root@pts00434-vm7 lp_original]# cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.1 (Maipo)
[root@pts00434-vm7 lp_original]# lscpu
Architecture:          ppc64le
Byte Order:            Little Endian
CPU(s):                8
On-line CPU(s) list:   0-7
Thread(s) per core:    1
Core(s) per socket:    8
Socket(s):             1
NUMA node(s):          1
Model:                 IBM pSeries (emulated by qemu)
L1d cache:             64K
L1i cache:             32K
NUMA node0 CPU(s):     0-7


Regards,
Amit

**Comment by superkhau** [comment]
The latest master is failing on my machine too. Unfortunately, I do not have the time to look into why yet, but feel free to submit a patch and I'll get it merged. The basic unit tests are failing ATM and we need to address this first.

**Comment by superkhau** [comment]
I think it may be related to https://github.com/strongloop/loopback-datasource-juggler/pull/815#issuecomment-173318761’

**Comment by ghatwala** [comment]
@superkhau , @jannyHou   - is your machine "ppc64le" architecture one ? Asking to confirm if these failures are observed on x86 machines as well.

Also let me know how/where to debug this issue , i am a newbie to both phantomjs/strongloop , i would be glad to contribute if i can debug it better ..

Regards,
Amit

**Comment by jannyHou** [comment]
Thank you @ghatwala my system is 64-bit windows, I haven't get your error but loopback master fails on some windows tests and I am trying to fix it. 
One warning I get from `npm install` regarding `karma` is:
loopback requires "karma": "^0.12.31" but this version seems incompatible with node 4.x and after.

**Comment by ghatwala** [comment]
@jannyHou , @superkhau  -  Were u guys able to test the loopback master , is it working at ur end ? .Please let me know if in case i can help debug further ?

**Comment by jannyHou** [comment]
thank you @ghatwala here is some update on testing result of loopback master:
https://github.com/strongloop/loopback/pull/1946
I am still testing it on windows. I hope I can reproduce the error on my local but unfortunately tests pass successfully on mine.

**Comment by ghatwala** [comment]
@jannyHou , @superkhau  - i installed nodejs from IBM sdk , now  below is my set up,
[root@pts00433-vm31 loopback]# node --version
v4.2.6
[root@pts00433-vm31 loopback]# phantomjs --version
2.0.0
I am getting the below errors . while i run  $ npm test command,
/*****************/
Running "mochaTest:unit" (mochaTest) task
 7 failing

  1) Checkpoint bumpLastSeq() and current() Should be no race condition for current() when calling in parallel:
     Uncaught AssertionError: expected [ Array(2) ] to have a length of 1 but got 2
      at test/checkpoint.test.js:38:32
      at node_modules/loopback-datasource-juggler/lib/dao.js:1519:9
      at done (node_modules/loopback-datasource-juggler/node_modules/async/lib/async.js:167:19)
      at node_modules/loopback-datasource-juggler/node_modules/async/lib/async.js:40:16
      at node_modules/loopback-datasource-juggler/lib/dao.js:1503:13
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:116:5)
      at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:91:8)
      at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:114:15)
      at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:91:8)
      at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:114:15)
      at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:91:8)

  2) Checkpoint bumpLastSeq() and current() Should be no race condition for bumpLastSeq() when calling in parallel:
     Uncaught AssertionError: expected [ Array(2) ] to have a length of 1 but got 2
      at test/checkpoint.test.js:54:32
      at node_modules/loopback-datasource-juggler/lib/dao.js:1519:9
      at done (node_modules/loopback-datasource-juggler/node_modules/async/lib/async.js:167:19)
      at node_modules/loopback-datasource-juggler/node_modules/async/lib/async.js:40:16
      at node_modules/loopback-datasource-juggler/lib/dao.js:1503:13
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at doNotify (node_modules/loopback-datasource-juggler/lib/observer.js:93:49)
      at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:116:5)
      at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:91:8)
      at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:114:15)
      at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:91:8)
      at Function.ObserverMixin._notifyBaseObservers (node_modules/loopback-datasource-juggler/lib/observer.js:114:15)
      at Function.ObserverMixin.notifyObserversOf (node_modules/loopback-datasource-juggler/lib/observer.js:91:8)

  3) loopback application pauses request stream during authentication:
     Uncaught TypeError: Cannot read property 'call' of undefined
      at Test.assert (node_modules/supertest/lib/test.js:221:5)
      at Server.assert (node_modules/supertest/lib/test.js:132:12)
      at emitCloseNT (net.js:1524:8)
      at node_modules/continuation-local-storage/node_modules/async-listener/glue.js:188:31

  4) RemoteConnector should support the save method:

      Uncaught AssertionError: false == true
      + expected - actual

      -false
      +true

      at HttpInvocation.<anonymous> (test/remote-connector.test.js:68:7)
      at node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:133:18
      at execStack (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:488:7)
      at RemoteObjects.execHooks (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:492:10)
      at HttpInvocation.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:131:15)
      at HttpInvocation.transformResponse (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:337:12)
      at Request._callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:261:10)
      at Request.self.callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:199:22)
      at Request.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:1036:10)
      at IncomingMessage.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:963:12)

  5) RemoteConnector Model Tests Model.create([data], [callback]) Create an instance of Model with given data and save to the attached data source:

      Uncaught AssertionError: false == true
      + expected - actual

      -false
      +true

      at HttpInvocation.<anonymous> (test/util/model-tests.js:136:11)
      at node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:133:18
      at execStack (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:488:7)
      at RemoteObjects.execHooks (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:492:10)
      at HttpInvocation.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:131:15)
      at HttpInvocation.transformResponse (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:337:12)
      at Request._callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:261:10)
      at Request.self.callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:199:22)
      at Request.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:1036:10)
      at IncomingMessage.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:963:12)

  6) RemoteConnector Model Tests model.updateAttributes(data, [callback]) Save specified attributes to the attached data source:
     Uncaught TypeError: user.updateAttributes is not a function
      at HttpInvocation.<anonymous> (test/util/model-tests.js:160:16)
      at node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:133:18
      at execStack (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:488:7)
      at RemoteObjects.execHooks (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:492:10)
      at HttpInvocation.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:131:15)
      at HttpInvocation.transformResponse (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:337:12)
      at Request._callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:261:10)
      at Request.self.callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:199:22)
      at Request.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:1036:10)
      at IncomingMessage.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:963:12)

  7) Replication over REST conflict resolution with model-level permissions as user with REPLICATE-only permissions allows reverse resolve() on the client:
     Uncaught TypeError: target.toObject is not a function
      at common/models/change.js:706:16
      at done (common/models/change.js:604:7)
      at node_modules/async/lib/async.js:251:17
      at done (node_modules/async/lib/async.js:132:19)
      at node_modules/async/lib/async.js:32:16
      at node_modules/async/lib/async.js:248:21
      at node_modules/async/lib/async.js:572:34
      at HttpInvocation.<anonymous> (common/models/change.js:598:9)
      at node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:133:18
      at execStack (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:488:7)
      at RemoteObjects.execHooks (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/remote-objects.js:492:10)
      at HttpInvocation.<anonymous> (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/rest-adapter.js:131:15)
      at HttpInvocation.transformResponse (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:337:12)
      at Request._callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/lib/http-invocation.js:261:10)
      at Request.self.callback (node_modules/loopback-connector-remote/node_modules/strong-remoting/node_modules/request/request.js:199:22)

Warning: Task "mochaTest:unit" failed. Use --force to continue.

Aborted due to warnings.
npm ERR! Test failed.  See above for more details.

/*****************/

Any pointers on above ?

**Comment by bajtos** [comment]
As part of our backlog grooming, we are closing issues that were inactive for a long period of time.

@ghatwala please open a new issue if the problem can be still reproduced on the latest loopback version.

**Comment by ghatwala** [comment]
@bajtos  : ok sure !!!
